### ISSUE TITLE
Support Implicit OIDC flow for the OIDC connector

### ISSUE THREAD

**Issue by colemickens** [coreos/dex#1254]
I know that Dex supports the implicit flow for the IdP, but I'd like to be able to use Dex with Portier, which only supports `response_type=id_token` and the `grant_type=implicit`.

Unfortunately, this requires tracking the `nonce` (which is required in the implicit flow) whereas the current OIDC connector seems stateless during the initial exchange.

Is there a pattern to follow here? Maybe a connector that has to stash some session data during the authentication flow? Is there a way to "start" the identity without it being treated as fully authenticated?

**Comment by colemickens** [comment]
I have this implemented in a fork by adding `ConnectorData` that can be persisted/accessed during the flow. But it's blocked on #1255.

**Comment by srenatus** [comment]
@colemickens Any update on this? It looks like #1255 isn't blocking this anymore.

**Comment by colemickens** [comment]
I had implemented it here. It stacks on another PR that isn't merged yet, though. https://github.com/dexidp/dex/pull/1256

**Comment by srenatus** [comment]
Sorry, sifting through so many issues and PRs. Thanks for the update.
